kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -65,6 +66,14 @@ class KyuubiSessionImpl(
   private var client: TCLIService.Client = _
   private var remoteSessionHandle: TSessionHandle = _
 
+  private val isClusterMode = sessionConf.getOption(SPARK_SUBMIT_DEPLOY_MODE)
+    .map(_.toLowerCase(Locale.ROOT)).contains("cluster")
+
+  if (isClusterMode) {
+    sessionConf.set(SPARK_YARN_MAX_APP_ATTEMPTS, "1")
+    sessionConf.set(SPARK_YARN_SUBMIT_WAIT_APP_COMPLETION, "false")
```

Comment:
Is it necessary?

Reply:
In cluster mode, clients should not be resident.

Reply:
what is wrong if we keep the client alive?

Reply:
No, but it consumes kyuubi service's resources.

Reply:
so i think set `spark.yarn.submit.waitAppCompletion=false` is necessary.

Reply:
Let us keep it configurable in case of any regression happen?

Reply:
ok